src/lib/libtls/tls_server.c

```diff
@@ -106,14 +115,31 @@ tls_accept_socket(struct tls *ctx, struct tls **cctx, int socket)
 			goto err;
 		}
 
-		if (SSL_set_fd(conn_ctx->ssl_conn, socket) != 1) {
-			tls_set_error(ctx, "ssl set fd failure");
-			goto err;
+		if (tls_config_is_dtls(ctx->config)) {
+			BIO *bio = BIO_new_dgram(socket, BIO_NOCLOSE);
+			SSL_set_bio(conn_ctx->ssl_conn, bio, bio);
+		} else {
+			if (SSL_set_fd(conn_ctx->ssl_conn, socket) != 1) {
+				tls_set_error(ctx, "ssl set fd failure");
+				goto err;
+			}
 		}
+		
 		SSL_set_app_data(conn_ctx->ssl_conn, conn_ctx);
 	}
 
-	if ((ret = SSL_accept(conn_ctx->ssl_conn)) != 1) {
+	if (tls_config_is_dtls(ctx->config)) {
+		union {
+			struct sockaddr_in s4;
+			struct sockaddr_in6 s6;
+		} client_addr = { .s6 = {} };
+		ret = DTLSv1_listen(conn_ctx->ssl_conn, &client_addr);
```

Comment:
Actually, a simple `SSL_accept()` should work here, too. But calling `DTLSv1_listen()` has some additional impact on packet counting.